content/4_Gold/BFS.mdx

```diff
@@ -254,6 +254,160 @@ deque.removeLast(); // [1, 3]
 
 ### Solution - Message Route
 
+**Time Complexity:** $\mathcal{O}(V+E)$
+
+We can observe is that there are many possible shortest paths to output. Fortunately, the problem states that we can print any valid solution. Notice that like every other 0/1 BFS problem, the distance of each node increases by 1 when we travel to the next level of unvisited nodes. However, the problem requires that we add additional information - in this case, the path. When we traverse from node $a$ to $b$, we can set the parent of $b$ to $a$. After the BFS is complete, this allows us to backtrack through the parents which ultimately leads us to our starting node. We know to terminate at node $1$ because its parent is 0. If there is no path to our end node, then the distance will not be reduced; it will remain as INT_MAX.
```

Comment:
```suggestion
We can observe is that there are many possible shortest paths to output. Fortunately, the problem states that we can print any valid solution. Notice that like every other 0/1 BFS problem, the distance of each node increases by 1 when we travel to the next level of unvisited nodes. However, the problem requires that we add additional information - in this case, the path. When we traverse from node $a$ to $b$, we can set the parent of $b$ to $a$. After the BFS is complete, this allows us to backtrack through the parents which ultimately leads us to our starting node. We know to terminate at node $1$ because its parent is $0$. If there is no path to our end node, then its distance will be infinity (or `INT_MAX`).
```

Reply:
You can change a bit (also should INT_MAX link to https://en.cppreference.com/w/c/types/limits ?)

Reply:
i linked it - can't hurt.